src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1332,6 +1333,10 @@ private function addWebLinkSection(ArrayNodeDefinition $rootNode, callable $enab
 
     private function addMessengerSection(ArrayNodeDefinition $rootNode, callable $enableIfStandalone)
     {
+        $defaultSerializer = class_exists(AmazonSqsSerializer::class)
```

Comment:
Hum, I missed this part: changing the serializer service for all transports.
I don't think that's the correct approach.
Instead, I see two ways:
1. patch the native serializer
2. fix the encoding in AmazonSqsSender, just after `$encodedMessage = $this->serializer->encode($envelope);`

Option 2. looks the most appropriate to me.

Reply:
Yeah, I wasn't sure about the best strategy for this.

The solution 1 leads to an impact for all the user of PHPSerializer even if they don't use AmazonSQS. There is no reason to follow AmazonSQS needs if you don't use it (so I tried to follow https://github.com/symfony/symfony/issues/49844#issuecomment-1766848137)

The solution 2 looks interesting, but if I add an extra base64_encode call after `$encodedMessage = $this->serializer->encode($envelope);`, how can I be sure it will be decoded by the serializer (since any custom Serializer can be used) ?

Reply:
> The solution 2 looks interesting, but if I add an extra base64_encode call, how can I be sure it will be decoded by the serializer (since any custom Serializer can be used) ?

If a custom serializer is used, it has to comply with Amazon's requirements, so this extra logic we're talking about shouldn't be triggered.

Reply:
> If a custom serializer is used, it has to comply with Amazon's requirements, so this extra logic we're talking about shouldn't be triggered.

Sure, we can go with this.

The other logic I had in mind was to add a Stamp to say 'I added an extra encoding' in order to reverse the encoding  in the AmazonSqsReceiver if the Stamp was here. Do you think it's better to add a stamp or we just go with the PR like this @nicolas-grekas ?

Reply:
I could add a Header to tell AmazonSqsReceiver it require an extra base64_decode.

WDYT @nicolas-grekas should I add the commit: https://github.com/VincentLanglet/symfony/compare/amazonSerializer...VincentLanglet:symfony:amazonSerializer-2?expand=1 or should I let the PR like this ?